src/ragged/_spec_set_functions.py

```diff
@@ -77,9 +105,24 @@ def unique_counts(x: array, /) -> tuple[array, array]:
 
     https://data-apis.org/array-api/latest/API_specification/generated/array_api.unique_counts.html
     """
-
-    x  # noqa: B018, pylint: disable=W0104
-    raise NotImplementedError("TODO 129")  # noqa: EM101
+    if isinstance(x, ragged.array):
+        if x.ndim == 0:
+            return unique_counts_result(
+                values=ragged.array([x]), counts=ragged.array([1])
+            )
+        elif len(x) == 1:
+            return unique_counts_result(
+                values=ragged.array(x), counts=ragged.array([1])
+            )
+        else:
```

Comment:
Same question for `x.ndim` as well as `len(x)`.

```suggestion
```

Reply:
Oh, if `x.ndim == 0`, then it doesn't have a layout, but the `x._impl` is the thing you want to pass to `np.unique`.